go/adbc/driver/snowflake/bulk_ingestion.go

```diff
@@ -334,21 +347,32 @@ func writeParquet(
 	parquetProps *parquet.WriterProperties,
 	arrowProps pqarrow.ArrowWriterProperties,
 ) error {
+
+	// don't start a new parquet file unless there is at least one record available
+	rec, ok := <-in
+	if !ok {
+		return ErrNoRecordsInStream
+	}
+
+	// initialize writer
 	pqWriter, err := pqarrow.NewFileWriter(schema, w, parquetProps, arrowProps)
 	if err != nil {
 		return err
 	}
 	defer pqWriter.Close()
 
-	var bytesWritten int64
-	for rec := range in {
-		if rec.NumRows() == 0 {
-			rec.Release()
-			continue
-		}
+	// write first record
+	bytesWritten, err := writeRecordToParquet(pqWriter, rec)
+	if err != nil {
+		return err
+	}
+	if targetSize > 0 && bytesWritten >= int64(targetSize) {
+		return nil
+	}
```

Comment:
can we add a comment here as to why we're doing an entire separate writing of the first record instead of letting it be handled by the channel and split?

Reply:
I revisited this and cleaned it up a bit. Between the refactor and added comments it should be more clear what's going on here.

Basically I wanted to return early and avoid any of the parquet file setup if there were no records left in the channel. Initially I grabbed the first record in order to check, but then I needed to write that to the file before I could continue to range over the remaining records.

Same idea after the refactor, just less duplication and explicitly initializing the parquet writer on the first loop iteration.

Reply:
I simplified this further. The parquet writer is always initialized, we just check the bytesWritten to see if we should discard the buffer or not.